### DISCUSSION TITLE
What is the alternative for `Signer.signMessage(arraify(message))`?

### DISCUSSION THREAD

**Discussion by ShookLyngs** [wevm/viem#697] [Question] [answered]
When I use `ethers` and I have to sign a hex, I can arraify the message and then sign it, like this:
```typescript
const sig = await signer.signMessage(arraify(message));
```

But in viem (wagmi), I cannot find an alternative of it to achive the feature I wanted.
I've tried `WalletClient.signMessage({ message: { raw: '0xxxx' } })`, but while signing the message, the message field in MetaMask's pupup showed '0x', how I called it:
```typescript
import { getAccount, getWalletClient } from 'wagmi/actions';

const walletClient = await getWalletClient();
const sig = await walletClient.signMessage({
  message: {
    raw: '0xce1619f2bb32f17f8d9dc5525d5974eed5213ce6e0986449678f8e477ea31d14',
  },
});
```

Can anyone help? Thanks.

**Comment by jxom** [comment, marked as answer]
“arrayifying” the hex is the same as just using the hex itself in `raw`.

**Reply by ShookLyngs** [reply]
I tried to pass `message.raw` like this (in wagmi):
```typescript
import { getAccount, getWalletClient } from 'wagmi/actions';

const walletClient = await getWalletClient();
const sig = await walletClient.signMessage({
  message: {
    raw: '0xce1619f2bb32f17f8d9dc5525d5974eed5213ce6e0986449678f8e477ea31d14',
  },
} as any);
```
But in MetaMask's popup, the message was just `0x`:
<img width="360" alt="image" src="https://github.com/wagmi-dev/viem/assets/44739165/a2cb79d1-8881-4e62-8d43-261174fe1f76">

**Reply by jxom** [reply]
What version of viem are you on?

**Reply by ShookLyngs** [reply]
> What version of viem are you on?

It works on the latest version, thanks for the tip!
